api.go

```diff
@@ -130,6 +131,74 @@ func (server *ApiServer) ProxyCreate(response http.ResponseWriter, request *http
 	}
 }
 
+func (server *ApiServer) Populate(response http.ResponseWriter, request *http.Request) {
+	input := []struct {
+		Proxy
+		Enabled *bool `json:"enabled"` // Overrides Proxy field to make field nullable
+	}{}
+
+	err := json.NewDecoder(request.Body).Decode(&input)
+	if apiError(response, joinError(err, ErrBadRequestBody)) {
+		return
+	}
+
+	// Check for valid input before creating any proxies
+	t := true
+	for i, p := range input {
+		if len(p.Name) < 1 {
+			apiError(response, joinError(fmt.Errorf("name"), ErrMissingField))
+			return
+		}
+		if len(p.Upstream) < 1 {
+			apiError(response, joinError(fmt.Errorf("upstream"), ErrMissingField))
```

Comment:
These errors don't say anything about the name of the proxy where this error occurs the way I read it. That may be unhelpful, what do you think?